tools/spice/spice_examples_partners.py

```diff
@@ -0,0 +1,289 @@
+
+
+def plot_data(data_file):
+    """Plot data from the output file.
+
+    Parameters
+    ----------
+    data_file: str
+        Path to the data file.
+    """
+    data = pd.read_csv(data_file, header=0, parse_dates=True, index_col="time_utc")
+
+    fig, axs = plt.subplots(3, 1, figsize=(10, 15), sharex=True)
+
+    date_str = data.index[0].strftime("%Y-%m-%d")
+    fig.suptitle(f"Data for {date_str}", fontsize=16, y=1.02)
+
+    # Plot for Azimuth
+    axs[0].plot(data.index, data["azimuth"], label="Azimuth", color="blue")
+    axs[0].set_ylabel("Degrees")
+    axs[0].set_title("Azimuth over Time")
+    axs[0].grid(True)
+
+    # Plot for Elevation
+    axs[1].plot(data.index, data["elevation"], label="Elevation", color="green")
+    axs[1].set_ylabel("Degrees")
+    axs[1].set_title("Elevation over Time")
+    axs[1].grid(True)
+
+    # Plot for Range to Target
+    axs[2].plot(data.index, data["range"], label="Range", color="red")
+    axs[2].set_ylabel("km")
+    axs[2].set_title("Range to Target over Time")
+    axs[2].grid(True)
+    axs[2].xaxis.set_major_formatter(mdates.DateFormatter("%H:%M"))
+
+    # Adjust the y-axis format
+    formatter = ScalarFormatter(useOffset=False)
+    formatter.set_scientific(True)
+    axs[2].yaxis.set_major_formatter(formatter)
+
+    plt.subplots_adjust(top=0.95)
+    plt.show()
+
+
+def main():
+    """Use main function."""
+    # Example usage for Longovilo-Hacienda, Chile
+    # https://www.findlatitudeandlongitude.com/l/LONGOVILO%2C+CHILE/
+    # Modify as needed
+    longitude = -71.41  # longitude in degrees
+    latitude = -33.94  # latitude in degrees
+    altitude = 0.157  # altitude in kilometers
+    start_utc_input = "2026 SEP 22 00:00:00"
+    stop_utc_input = "2026 SEP 22 23:59:59"
+    time_interval = int(1e3)  # seconds between data points
+
+    # Define the path to the kernel directory relative to this script
+    kernel_directory = Path(__file__).parents[0]
+    kernels = get_kernels(kernel_directory)
+
+    # Load spice kernels and perform calculations.
+    with (
+        spice.KernelPool(kernels),
+        open(
+            "../tests/test_data/spice_partner_example/spice_output_data.txt", "w"
+        ) as file,
+    ):
+        # Write the header line to the output file
+        file.write("time_et,time_utc,azimuth,elevation,range\n")
+
+        start_et_input = spice.utc2et(start_utc_input)
+        stop_et_input = spice.utc2et(stop_utc_input)
+
+        # Get time coverage from ephemeris kernel
+        get_coverage(
+            str(kernel_directory / "IMAP_launch20250429_1D.bsp"),
+            start_et_input,
+            stop_et_input,
+        )
+
+        # Get azimuth (degrees) and elevation (degrees) and
+        # range (km) to the target for each time step
+        for time in range(int(start_et_input), int(stop_et_input), time_interval):
+            # Convert to UTC time for readability
+            time_utc = spice.et2utc(time, "C", 0)
```

Comment:
I find the ISOC to be easier and more portable to read.

```suggestion
            time_utc = spice.et2utc(time, "ISOC", 0)
```

Options from: https://naif.jpl.nasa.gov/pub/naif/toolkit_docs/C/cspice/et2utc_c.html
```bash
                  "C"      "1986 APR 12 16:31:09.814"
                  "D"      "1986-102 // 16:31:12.814"
                  "J"      "JD 2446533.18834276"
                  "ISOC"   "1987-04-12T16:31:12.814"
                  "ISOD"   "1987-102T16:31:12.814"
```